File: Segregate0sAnd1s/Segregate0sAnd1s.py
# https://www.geeksforgeeks.org/problems/segregate-0s-and-1s5106/1


#User function Template for python3

class Solution:
    def segregate0and1(self, arr):
        lengthOfArr=len(arr)
        beginIndex=0
        countOfZeros=0
        for i in range(0, lengthOfArr, 1):
            if arr[i]==0:
                countOfZeros+=1
        while(countOfZeros>0):
            arr[beginIndex]=0
            beginIndex+=1
            countOfZeros-=1
        while(beginIndex<=lengthOfArr-1):
            arr[beginIndex]=1
            beginIndex+=1


#{ 
 # Driver Code Starts
class IntArray:

    def __init__(self) -> None:
        pass

    def Input(self, n):
        arr = [int(i) for i in input().strip().split()]  #array input
        return arr

    def Print(self, arr):
        for i in arr:
            print(i, end=" ")
        print()


if __name__ == "__main__":
    t = int(input())
    for _ in range(t):

        arr = list(map(int, input().strip().split()))

        obj = Solution()
        obj.segregate0and1(arr)

        print(*arr)

# } Driver Code Ends

File: Segregate0sAnd1s/Segregate0sAnd1s.cpp
/*
https://www.geeksforgeeks.org/problems/segregate-0s-and-1s5106/1
*/

//{ Driver Code Starts
// Initial template for C++
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
// User function template for C++

class Solution {
  public:
    void segregate0and1(vector<int> &arr) {
        int sz=arr.size(), idx=0;
        int countZeros=0;
        for(int i=0; i<sz; ++i){
            if(arr[i]==0) countZeros++;
        }
        while(countZeros--){
            arr[idx++]=0;
        }
        while(idx<=sz-1){
            arr[idx++]=1;
        }
    }
};

//{ Driver Code Starts.
int main() {
    int t;

    cin >> t;
    cin.ignore();
    while (t--) {
        vector<int> arr;
        string input;
        getline(cin, input);
        stringstream ss(input);
        int number;
        while (ss >> number) {
            arr.push_back(number);
        }
        int n = arr.size();
        Solution ob;
        ob.segregate0and1(arr);
        for (int i = 0; i < n; i++) {
            cout << arr[i] << " ";
        }
        cout << "\n";
    }
    return 0;
}
// } Driver Code Ends